CRM/Contribute/BAO/Contribution.php

```diff
@@ -3663,7 +3676,7 @@ public static function updateFinancialAccounts(&$params, $context = NULL) {
       }
     }
 
-    CRM_Core_BAO_FinancialTrxn::createDeferredTrxn(CRM_Utils_Array::value('line_item', $params), $params['contribution'], TRUE, $context);
+    CRM_Core_BAO_FinancialTrxn::createDeferredTrxn(CRM_Utils_Array::value('line_item', $params), $params['id'], TRUE, $context);
```

Comment:
@monishdeb tests are complaining about id being undefined here

Reply:
Hmm, thanks for pointing that out .. will fix it, also this patch needs some additional work, so I marked with [WIP]